src/components/network-modification-tree-pane.js

```diff
@@ -260,6 +260,25 @@ export const NetworkModificationTreePane = ({
                     });
                     broadcastChannel.postMessage(noSelectionForCopy);
                 }
+            } else if (
+                studyUpdatedForce.eventData.headers['updateType'] ===
+                'nodeRenamed'
+            ) {
+                updateNodes([studyUpdatedForce.eventData.headers['node']]);
+            } else if (
+                studyUpdatedForce.eventData.headers['updateType'] ===
+                'nodeBuildStatusUpdated'
+            ) {
+                updateNodes(studyUpdatedForce.eventData.headers['nodes']);
+                if (
+                    studyUpdatedForce.eventData.headers['nodes'].some(
+                        (nodeId) => nodeId === currentNodeRef.current?.id
+                    )
+                ) {
+                    dispatch(
+                        removeNotificationByNode([currentNodeRef.current?.id])
+                    );
+                }
```

Comment:
this method (useEffect) is becoming more and more complicated and contains a lot of if/else blocks (100 lines of code)
I think we should think about refactoring this method

Reply:
Yes
I have already noted this point in the  notification refactoring us